test/test_helper.rb

```diff
@@ -5,14 +5,20 @@
 require 'rubocop'
 
 module TestHelper
-  def assert_offense(violation_message)
-    refute_empty(cop.offenses, "Expected a #{cop.cop_name} offense to be detected but there was none")
+  def assert_offense(cop_name = nil, violation_message = nil)
+    matching_offenses = cop_name.nil? ? cop.offenses : cop.offenses.select { _1.cop_name == cop_name }
+    message = ['Expected an offense', 'to be detected but there was none']
+    message.insert(1, "named #{cop_name}") if cop_name
+    message_string = message.join(' ')
```

Comment:
Will show a generic message if the cop name is not specified, and a specific one if it is

`'Expected an offense to be detected but there was none'` vs `'Expected an offense named Custom/NoMagicNumbers to be detected but there was none`